google-common/src/main/scala/org/apache/pekko/stream/connectors/google/ResumableUpload.scala

```diff
@@ -61,14 +61,9 @@ private[connectors] object ResumableUpload {
 
         val in = Flow[ByteString]
           .via(chunker(uploadChunkSize))
-          .statefulMapConcat { () =>
-            var cumulativeLength = 0L
-            bytes => {
-              val chunk = Chunk(bytes, cumulativeLength)
-              cumulativeLength += bytes.length
-              chunk :: Nil
-            }
-          }
+          .statefulMap(() => 0L)((cumulativeLength, bytes) =>
+              (cumulativeLength + bytes.length, Chunk(bytes, cumulativeLength)),
+            _ => None)
```

Comment:
These three flow can be optimized with a single statefulmap I think ,but leave that for another pr.